test/integration/model/json.test.js

```diff
@@ -227,7 +227,7 @@ describe(Support.getTestDialectTeaser('Model'), function () {
           return this.Event.findAll({
             where: {
               data: {
-                employment: null
+                employment: current.dialect.name === 'mysql' ? 'null' : null
```

Comment:
It's not the sequelize JSON processing that is returning `'null'` as a string. Let's say we have the following JSON data:
```json
{ "test": null }
```

In that case, MySQL returns `'null'` as a string when we run the following query:
```sql
SELECT data->>'$.test' FROM table
```

MySQL only returns NULL when the property doesn't exist in the JSON document.
I'm thinking about using the [JSON_TYPE function](https://dev.mysql.com/doc/refman/5.7/en/json-attribute-functions.html#function_json-type) because, in that case, it returns `'NULL'` (rather than `'STRING'`).
So the final query would be something similiar to this:
```sql
SELECT * FROM table WHERE (data->>'$.test' = NULL OR JSON_TYPE(data->>'$.test') = 'NULL')
```